lua/telescope/previewers.lua

```diff
@@ -34,6 +38,38 @@ end
 
 local add_quotes = valuate_shell()
 
+local function readFile(path, callback)
+  vim.loop.fs_open(path, "r", 438, function(err, fd)
+    assert(not err, err)
+    vim.loop.fs_fstat(fd, function(err, stat)
+      assert(not err, err)
+      vim.loop.fs_read(fd, stat.size, 0, function(err, data)
+        assert(not err, err)
+        vim.loop.fs_close(fd, function(err)
+          assert(not err, err)
+          return callback(data)
+        end)
+      end)
+    end)
+  end)
+end
+
+local filetype_cache = vim.fn.split(vim.fn.execute('autocmd filetypedetect'), '\n')
+local determine_filetype = function(filepath)
+  local ext = vim.fn.fnamemodify(filepath, ':e')
+  if ext == '' or ext == nil then
+    return ''
+  end
+  local matchings = vim.fn.uniq(vim.fn.sort(vim.tbl_filter(function(val)
+    return val:match('.*%.' .. ext .. '%s')
+  end, filetype_cache)))
```

Comment:
This converts back and forth a bunch between vimscript and lua. Not good for loading times.